apptainer/run_example_cpu_job.sh

```diff
@@ -0,0 +1,35 @@
+# salloc --time=0:10:0 --mem-per-cpu=4G --ntasks=4 --account=def-mbowling
+
+IMAGE_NAME=${1:-minigrid_pytorch_2.4.0-python3.12.5-devell.sif}
+
+# Load apptainer and copy over the container
+module load apptainer/1.2.4
+cp ~/projects/def-mbowling/aorenste/apptainer_images/$IMAGE_NAME $SLURM_TMPDIR/
+cd $SLURM_TMPDIR/
+
+# Set up SOCKS5 proxy
+echo "Setting up SOCKS5 proxy..."
+ssh -q -N -T -f -D 8888 `echo $SSH_CONNECTION | cut -d " " -f 3`
+export ALL_PROXY=socks5h://localhost:8888
+
+# clone an example repo
+git config --global http.proxy $ALL_PROXY
+git clone --quiet https://github.com/AdrianOrenstein/docker-pytorch-template.git $SLURM_TMPDIR/project
+
+# sanity check we have access to the pip packages
+apptainer run \
+    --env ALL_PROXY=$ALL_PROXY \
+    --env APPEND_PATH=$SLURM_TMPDIR/project \
+    $SLURM_TMPDIR/$IMAGE_NAME python -c "import torch, minigrid; print(f'{torch.__version__=}\t{minigrid.__version__=}')"
+
+# check that we can use torch on the CPU
+apptainer run \
+    --env ALL_PROXY=$ALL_PROXY \
+    --env APPEND_PATH=$SLURM_TMPDIR/project \
+    $SLURM_TMPDIR/$IMAGE_NAME python -c "import torch; a=torch.ones(50).cpu(); out=(a+a)*2; print(out.shape, out)"
+
+# check that we can run python modules on the cloned repo
+apptainer run \
+    --env ALL_PROXY=$ALL_PROXY \
+    --env APPEND_PATH=$SLURM_TMPDIR/project \
+    $SLURM_TMPDIR/$IMAGE_NAME python -m black $SLURM_TMPDIR/project
```

Comment:
I wonder if we can make these actual pytests? Python.subprocess?

Reply:
[done](https://github.com/AdrianOrenstein/docker-pytorch-template/pull/5/commits/fed9f2fe8a7c4229ab9152b107f2af5f8bd17fd9#diff-9cc1ac48784152634df4367b45d2e731aa186f39b99148030123ce07a6af39ed)